src/hotspot/share/opto/vtransform.cpp

```diff
@@ -144,6 +144,224 @@ void VTransformApplyResult::trace(VTransformNode* vtnode) const {
 }
 #endif
 
+// We use two comparisons, because a subtraction could underflow.
+#define RETURN_CMP_VALUE_IF_NOT_EQUAL(a, b) \
+  if (a < b) { return -1; }                 \
+  if (a > b) { return  1; }
+
+// Helper-class for VTransformGraph::has_store_to_load_forwarding_failure.
+// It represents a memory region: [ptr, ptr + memory_size)
+class VPointerRecord : public StackObj {
```

Comment:
Actually, it is supposed to be a `VPointer`, but one where we can give some `iv_offset`, i.e. mutate the offset. But ok, I could rename it to `VMemoryRegion`, it would not hurt :)

Reply:
done